src/Features/Core/Portable/GoToDefinition/IFindDefinitionService.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Navigation;
+
+namespace Microsoft.CodeAnalysis.GoToDefinition
+{
+    internal interface IFindDefinitionService : ILanguageService
```

Comment:
1. Why do we need to extract this?
2. This may be implemented by ts/f#. So this may break them.

Reply:
> 1. Why do we need to extract this?

To remove LSP's dependency on EditorFeatures.

> 2. This may be implemented by ts/f#. So this may break them.

Well, F# I would expect to be using the external access layer? I'm also not really sure how this would break them: the existing interface they would have been using would still have these same methods, so I would expect a recompile would just work.

Reply:
Problem is if they're implementing these interfaces explicitly.

Reply:
Problem is that we wouldn't see their impl as an impl of this new interface

Reply:
Also, deriving interface from interface is not something we generally do. Would prefer two separate interfaces

Reply:
We can add the new interfaces at the lower layer. Then provide bridges from the old interfaces to the new ones

Reply:
@CyrusNajmabadi readded the deleted method to `IGoToDefinitionService` to avoid breaking consumers. The implementation still inherits from the base type to provide the impl of that method, but it should avoid TS and F# breaks.